linalg/solvers.cpp

```diff
@@ -1319,6 +1319,54 @@ void NewtonSolver::Mult(const Vector &b, Vector &x) const
    final_norm = norm;
 }
 
+void NewtonSolver::CheckJacobian(Vector &x, Array<int> &ess_tdof)
+{
+   MFEM_ASSERT(oper != nullptr, "Call SetOperator first.");
+
+   double eps = 1e-8;
+   Vector xpert(height), one(height), fx(height),
```

Comment:
You should check the sizes here. If the method belongs to the Operator class, then it may be the case that width != height. The Frechet derivative you are testing against is still well defined.